lv_objx/lv_tabview.c

```diff
@@ -851,41 +868,77 @@ static void tabview_realign(lv_obj_t * tabview)
-				break;
-		}
+
+        /* Buttons top or bottom */
+        if(ext->btns_pos <= LV_TABVIEW_BTNS_POS_BOTTOM)
+        {
+			/*Set the indicator widths*/
+			lv_coord_t indic_width = (lv_obj_get_width(tabview) - style_btn_bg->body.padding.inner * (ext->tab_cnt - 1) -
+									  2 * style_btn_bg->body.padding.hor) / ext->tab_cnt;
+			lv_obj_set_width(ext->indic, indic_width);
+
+			/*Set the tabs height*/
+			lv_coord_t btns_height = lv_font_get_height(style_btn_rel->text.font) +
+									 2 * style_btn_rel->body.padding.ver +
+									 2 * style_btn_bg->body.padding.ver;
+			lv_obj_set_height(ext->btns, btns_height);
+
+			lv_obj_set_height(ext->content, lv_obj_get_height(tabview) - lv_obj_get_height(ext->btns));
+
+			switch(ext->btns_pos) {
+				case LV_TABVIEW_BTNS_POS_TOP:
+					lv_obj_align(ext->btns, NULL, LV_ALIGN_IN_TOP_LEFT, 0, 0);
+					lv_obj_align(ext->content, ext->btns, LV_ALIGN_OUT_BOTTOM_LEFT, 0, 0);
+					break;
+				case LV_TABVIEW_BTNS_POS_BOTTOM:
+					lv_obj_align(ext->content, NULL, LV_ALIGN_IN_TOP_LEFT, 0, 0);
+					lv_obj_align(ext->btns, ext->content, LV_ALIGN_OUT_BOTTOM_LEFT, 0, 0);
+					break;
+			}
+        }
+        /* Buttons left or right */
+        else if(ext->btns_pos > LV_TABVIEW_BTNS_POS_BOTTOM)
+        {
+			/*Set the indicator size*/
+			lv_coord_t indic_height = (lv_obj_get_height(tabview) - style_btn_bg->body.padding.inner * (ext->tab_cnt - 1) -
+									  2 * style_btn_bg->body.padding.hor) / ext->tab_cnt;
+
+			lv_obj_set_width(ext->indic, lv_obj_get_height(ext->indic));
+			lv_obj_set_height(ext->indic, indic_height);
+
+        	/*Set tabs buttons size*/
+			lv_obj_set_height(ext->btns, lv_obj_get_height(tabview));
+			lv_obj_set_width(ext->btns, 70);
+
+        	/*Set content size*/
+			lv_obj_set_size(ext->content, lv_obj_get_width(tabview) - lv_obj_get_width(ext->btns), lv_obj_get_height(tabview));
+
+			switch(ext->btns_pos) {
+				case LV_TABVIEW_BTNS_POS_LEFT:
+					lv_obj_align(ext->btns, NULL, LV_ALIGN_IN_TOP_LEFT, 0, 0);
+					lv_obj_align(ext->content, ext->btns, LV_ALIGN_OUT_RIGHT_TOP, 0, 0);
+					break;
+				case LV_TABVIEW_BTNS_POS_RIGHT:
+					lv_obj_align(ext->btns, NULL, LV_ALIGN_IN_TOP_RIGHT, 0, 0);
+					lv_obj_align(ext->content, ext->btns, LV_ALIGN_OUT_LEFT_TOP, 0, 0);
+					break;
+			}
+        }
     }
 
     lv_obj_t * pages = lv_obj_get_child(ext->content, NULL);
     while(pages != NULL) {
         if(lv_obj_get_signal_func(pages) == tabpage_signal) {  /*Be sure adjust only the pages (user can other things)*/
-            lv_obj_set_size(pages, lv_obj_get_width(tabview), lv_obj_get_height(ext->content));
+            lv_obj_set_size(pages, lv_obj_get_width(ext->content), lv_obj_get_height(ext->content));
         }
         pages = lv_obj_get_child(ext->content, pages);
     }
 
     if(!ext->btns_hide) {
-    	lv_obj_align(ext->indic, ext->btns, LV_ALIGN_IN_BOTTOM_LEFT, 0, 0);
+    	if(ext->btns_pos <= LV_TABVIEW_BTNS_POS_BOTTOM)
```

Comment:
Again, I think an explicit `btns_pos == LV_TABVIEW_BTNS_POS_BOTTOM || btns_pos == LV_TABVIEW_BTNS_POS_TOP` would be better. It's easier to read and understand.